server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/InMemoryTreeNode.java

```diff
@@ -71,4 +97,69 @@ public byte getNodeType() {
         return nodeType;
     }
 
+    public boolean hasChild() {
+        return !(this.children.isEmpty() && this.childStarNode.get() == null);
+    }
+
+    public int getDimensionId() {
+        return dimensionId;
+    }
+
+    public int getStartDocId() {
+        return startDocId;
+    }
+
+    public int getEndDocId() {
+        return endDocId;
+    }
+
+    public void setNodeType(byte nodeType) {
+        this.nodeType = nodeType;
+    }
+
+    public void addChildNode(InMemoryTreeNode childNode, Long dimensionValue) {
+        if (childNode.getNodeType() == StarTreeNodeType.STAR.getValue()) {
+            assert this.childStarNode.get() == null;
```

Comment:
sorry i know we discussed this but since u already have setonce do we really need assertion ?

Reply:
we can remove the assertion!